### PR TITLE
Create proc_creation_macos_jxa_in-memory_execution.yml

### PR DESCRIPTION
**Pull request by D4rkCiph3r**
macOS Rule for T1059.007
JXA in-memory execution via OSAScript

### CODE REVIEW

**Comment by nasbench** [comment]
Thanks again.

I've updated the rule logic a little bit to be more readable. And added space to the ` -e ` and ` -l ` arguments, so they are not short atoms and impact performance for some backends (in theory).

Please check :)

Also, there is a certain order of fields that we follow. Basically, grab a similar rule and modify it and it should be all good.

**Comment by nasbench** [comment]
@D4rkCiph3r Please follow the same order of fields as the previous rule. Much appreciated :D

**Review by frack113 — APPROVED** [review]
LGTM

File: rules/macos/process_creation/proc_creation_macos_jxa_payoad_execution.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+title: JXA In-memory Execution Via OSAScript
```
Comment (frack113) [inline rules/macos/process_creation/proc_creation_macos_jxa_payoad_execution.yml:1]:
```suggestion
title: JavaScript File Execution Via OSAScript
```

Comment (nasbench) [inline rules/macos/process_creation/proc_creation_macos_jxa_payoad_execution.yml:1]: No need for this rule. Just add the `.js` extension similar to how you added `.scpt` in #3993 to the same rules in another PR (or simply pull from master and update this)

**Review by nasbench — DISMISSED** [review]

**Pull request merged**